test/org/sagebionetworks/bridge/validators/SubpopulationValidatorTest.java

```diff
@@ -65,9 +68,39 @@ public void testValidation() {
             assertMessage(e, "noneOfGroups", " 'wrongGroup' is not in enumeration");
             assertMessage(e, "allOfSubstudyIds", " 'substudyC' is not in enumeration");
             assertMessage(e, "noneOfSubstudyIds", " 'substudyD' is not in enumeration");
+            assertMessage(e, "dataGroupsAssignedWhileConsented", " 'dataGroup3' is not in enumeration: group1, group2");
+            assertMessage(e, "substudyIdsAssignedOnConsent", " 'substudyC' is not in enumeration: substudyA, substudyB");
         }
     }
     
+    @Test
+    public void emptyListsOK() {
+        Subpopulation subpop = Subpopulation.create();
+        subpop.setStudyIdentifier("test-study");
+        subpop.setGuidString("AAA");
+        subpop.setName("Name");
+        subpop.setDataGroupsAssignedWhileConsented(ImmutableSet.of());
+        subpop.setSubstudyIdsAssignedOnConsent(ImmutableSet.of());
+        
+        Validate.entityThrowingException(validator, subpop);
+        assertTrue(subpop.getDataGroupsAssignedWhileConsented().isEmpty());
+        assertTrue(subpop.getSubstudyIdsAssignedOnConsent().isEmpty());
```

Comment:
We also need a test with valid non-empty lists.

Reply:
Added to the valid subpopulation test.